### PR TITLE
 Simplify hex2str, remove broken assertion causing exec regression.

### PR DESCRIPTION
**Pull request by donporter**
Fix a regression on exec in trying to repro #59. Remove an assertion that doesn't seem to be doing much good, given that the size is always statically determined in hex2str.

Add the first test case mentioned in #59 to CI (which seems to pass now).

---
This change is [Reviewable](https://reviewable.io/reviews/oscarlab/graphene/215)

### CODE REVIEW

File: Pal/src/host/Linux-SGX/enclave_framework.c [hunk 1]
```diff
@@ -901,7 +901,8 @@ int init_enclave (void)
 
 int _DkStreamKeyExchange (PAL_HANDLE stream, PAL_SESSION_KEY * keyptr)
 {
-    unsigned char session_key[32] __attribute__((aligned(32)));
+    uint8_t session_key[sizeof(PAL_SESSION_KEY)]
+        __attribute__((aligned(sizeof(PAL_SESSION_KEY))));
```
[the remark this answers is not in the document]
Comment (donporter) [reply Pal/src/host/Linux-SGX/enclave_framework.c]: +1

**Comment by donporter** [comment]
I'm good with the changes you made.

**Comment by chiache** [comment]
@donporter Ready to merge if this looks good to you.

**Comment by chiache** [comment]
Test this please

**Comment by chiache** [comment]
Test this please

**Comment by chiache** [comment]
Looks like the 18.04 tests have made the original tests easy to timeout.

**Comment by chiache** [comment]
test this please

**Comment by donporter** [comment]
We can ignore the 18.04 jobs - 18.04 support is still in progress.

**Comment by donporter** [comment]
Test this please

**Comment by donporter** [comment]
Can I get a final approval on the squashed/rebased code?

**Pull request merged**